components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -167,7 +167,11 @@ public void invoke(Request request, Response response) throws IOException, Servl
             log.error("Identity Runtime Exception occurred in Authentication valve :", e);
             APIErrorResponseHandler.handleErrorResponse(authenticationContext, response,
                     HttpServletResponse.SC_SERVICE_UNAVAILABLE, null);
-        } catch (URISyntaxException | PatternSyntaxException e) {
+        } catch (URISyntaxException e) {
+            log.debug("Invalid URI syntax of the request :", e);
```

Comment:
```suggestion
            log.debug("Invalid URI syntax of the request: ", e);
```

Reply:
Also check for isDebugEnabled

Reply:
In this case we may not need as we don't do any costly string concatenation etc..
https://www.igorkromin.net/index.php/2015/05/12/are-guard-statements-like-isdebugenabled-necessary-when-using-log4j/